File: backend/src/models/EntradaDiario.js
'use strict';

const { CategoriaDiario } = require('./enums');

// Entidade EntradaDiario (<<entity>>) - UC004.
class EntradaDiario {
  constructor({
    id,
    grupoId,
    categoria,
    descricao,
    importante = false,
    autorId,
    autorNome = null,
    criadaEm = new Date(),
  }) {
    this.id = id;
    this.grupoId = grupoId;
    this.categoria = categoria;
    this.descricao = descricao;
    this.importante = importante;
    this.autorId = autorId; // nao alteravel apos salvar (FE/passo 5)
    this.autorNome = autorNome;
    this.criadaEm = criadaEm; // preenchida automaticamente, nao alteravel
  }

  toJSON() {
    return { ...this };
  }
}

EntradaDiario.Categoria = CategoriaDiario;

module.exports = EntradaDiario;

File: app/lib/models/entrada_diario.dart
/// Modelo de Entrada do Diario de Saude (UC004).
class EntradaDiario {
  final String id;
  final String grupoId;
  final String categoria;
  final String descricao;
  final bool importante;
  final String? autorNome;
  final DateTime criadaEm;

  EntradaDiario({
    required this.id,
    required this.grupoId,
    required this.categoria,
    required this.descricao,
    required this.importante,
    required this.autorNome,
    required this.criadaEm,
  });

  factory EntradaDiario.fromJson(Map<String, dynamic> j) => EntradaDiario(
        id: j['id'],
        grupoId: j['grupoId'] ?? '',
        categoria: j['categoria'] ?? '',
        descricao: j['descricao'] ?? '',
        importante: j['importante'] ?? false,
        autorNome: j['autorNome'],
        criadaEm: DateTime.tryParse(j['criadaEm']?.toString() ?? '') ?? DateTime.now(),
      );
}
